### PR TITLE
feat: Convert notification center to TS

### PR DESCRIPTION
**Pull request by yavorona**
## Summary
- Convert `notification_center` module to TS

## Test plan

Existing unit tests

## Issues
[OASIS-7067](https://optimizely.atlassian.net/browse/OASIS-7067)

### CODE REVIEW

File: packages/optimizely-sdk/lib/core/notification_center/index.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+   * If there was an error and the listener was not added, addNotificationListener returns -1. This
+   * can happen if the first argument is not a valid notification type, or if the same callback
+   * function was already added as a listener by a prior call to this function.
+   */
+  addNotificationListener<T extends ListenerPayload>(
+    notificationType: string,
+    callback: NotificationListener<T>
+  ): number {
+    try {
+      const isNotificationTypeValid = objectValues(NOTIFICATION_TYPES ).indexOf(notificationType as notificationTypesEnum) > -1;
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/core/notification_center/index.ts:89]: Not requesting change, but this is coming up a lot. I wonder if there's a more idiomatic solution for TS that doesn't require using `as`  when iterating over `Object.keys()/Object.values()` return value.

**Review by mjc1283 — APPROVED** [review]
LGTM. Brief comments added.

File: packages/optimizely-sdk/lib/core/notification_center/index.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+} from '../../utils/enums';
+
+const MODULE_NAME = 'NOTIFICATION_CENTER';
+
+export interface NotificationCenterOptions {
+  logger: LogHandler;
+  errorHandler: ErrorHandler;
+}
+
+export interface ListenerEntry {
```
[the remark this answers is not in the document]
Comment (yavorona) [reply packages/optimizely-sdk/lib/core/notification_center/index.ts:34]: No, will be updated.

File: packages/optimizely-sdk/lib/core/notification_center/index.ts (hunk 1 above)
Comment (yavorona) [reply packages/optimizely-sdk/lib/core/notification_center/index.ts]: Good point. Not sure how to avoid using it here, will look into it.

**Pull request merged**